source/MaterialXTest/ShaderValid.cpp

```diff
@@ -706,97 +692,56 @@ TEST_CASE("MaterialX documents", "[shadervalid]")
+            {
+                docValidLog << validationErrors << std::endl;
+            }
+            CHECK(validDoc);
+
+            std::vector<mx::TypedElementPtr> elements;
+            mx::findRenderableElements(doc, elements);
+
+            std::string outputPath = mx::FilePath(dir) / mx::FilePath(mx::removeExtension(file));
+            for (auto element : elements)
+            { 
+                mx::OutputPtr output = element->asA<mx::Output>();
+                mx::ShaderRefPtr shaderRef = element->asA<mx::ShaderRef>();
+                mx::NodeDefPtr nodeDef = nullptr;
+                if (output)
                 {
-                    docValidLog << validationErrors << std::endl;
+                    nodeDef = output->getConnectedNode()->getNodeDef();
                 }
-                CHECK(validDoc);
-
-                std::unordered_set<mx::OutputPtr> shaderrefOutputs;
-                for (auto material : materials)
+                else if (shaderRef)
                 {
-                    for (auto shaderRef : material->getShaderRefs())
-                    {
-                        if (!shaderRef->hasSourceUri())
-                        {
-                            std::string outputPath = mx::FilePath(dir) / mx::FilePath(mx::removeExtension(file));
-                            mx::string elementName = mx::replaceSubstrings(shaderRef->getNamePath(), pathMap);
-#ifdef MATERIALX_BUILD_GEN_GLSL
-                            runGLSLValidation(elementName, shaderRef, *glslValidator, *glslShaderGenerator, lightHandler, doc, glslLog, false, outputPath);
-#endif
-#ifdef MATERIALX_BUILD_GEN_OSL
-                            runOSLValidation(elementName, shaderRef, *oslValidator, *oslShaderGenerator, doc, oslLog, false, outputPath);
-#endif
-
-                            // Find all bindinputs which reference outputs and outputgraphs
-                            for (auto bindInput : shaderRef->getBindInputs())
-                            {
-                                mx::OutputPtr outputPtr = bindInput->getConnectedOutput();
-                                if (outputPtr)
-                                {
-                                    shaderrefOutputs.insert(outputPtr);
-                                }
-                            }
-                        }
-                    }
+                    nodeDef = shaderRef->getNodeDef();
                 }
-
-                // Find node graph outputs
-                for (mx::NodeGraphPtr nodeGraph : nodeGraphs)
+                if (nodeDef)
                 {
-                    // Skip anything from an include file including libraries.
-                    if (!nodeGraph->hasSourceUri())
+                    mx::string elementName = mx::replaceSubstrings(element->getNamePath(), pathMap);
+#ifdef MATERIALX_BUILD_GEN_GLSL
+                    if (nodeDef->getImplementation(glslShaderGenerator->getTarget(), glslShaderGenerator->getLanguage()))
```

Comment:
Implementation specific check is left outside of finding if something is renderable.